core/tests/test_utils.py

```diff
@@ -3030,12 +3033,35 @@ def get_response_without_checking_for_errors(
                 params, dict,
                 msg='Expected params to be a dict, received %s' % params)
 
+            if http_method != 'GET':
```

Comment:
Bit confused about this. I imagine you should just have an assertion at the outset that http_method is one of GET, POST, PUT, DELETE. Why should it be nested in "if params is not None"?

Also, is there some relation that should exist between the http_method and params (e.g. do some methods not support params)? If so we should probably check for that as well.

Reply:
Added assertion that http_method is one of GET, POST, PUT, DELETE and unindented it from under if params is not none. Done.